dipy/align/imaffine.py

```diff
@@ -144,6 +144,8 @@ def set_affine(self, affine):
             self.affine_inv = None
             return
         try:
+            if np.isnan(np.sum(affine)):
```

Comment:
Surely these two lines should go above the `try` block?   I think the `np.sum` trick is to avoid large temporary arrays, for me a simple `np.any(np.isnan(x))` is faster still, and I think it's easier to read.